src/transformStateWithClones.js

```diff
@@ -8,6 +8,40 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const arrayWithObjects = [];
+  let cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        cloneState = {
+          ...cloneState,
+          ...action.extraData,
+        };
+
+        break;
+
+      case 'removeProperties':
+        cloneState = removeProperties(cloneState, action.keysToRemove);
```

Comment:
The `removeProperties` function modifies the `newObject` directly, which goes against the checklist item stating that objects or arrays should not be mutated. Instead, create a copy of `newObject` within the `removeProperties` function and perform deletions on that copy.